pontoon/base/static/js/translate.js

```diff
@@ -15,25 +15,74 @@ var Pontoon = (function (my) {
      */
     setFilter: function (type) {
       type = type || 'all';
-      var node = $('#filter .menu li[data-type="' + type + '"]');
+      var node = $('#filter .menu [data-type="' + type + '"]'),
+          selectorType = type,
+          isTimeRange = false,
+          isInvalid = false;
 
       // Validate filter
       if (node.length === 0) {
-        type = 'all';
-        node = $('#filter .menu li[data-type="' + type + '"]');
+        // Special case: filter by time range
+        if (/^[0-9]{12}-[0-9]{12}$/.test(type)) {
+          // Populate from/to inputs if filter specified in URL
+          if (!$('#from').val() && !$('#to').val()) {
+            var from = new Date(
```

Comment:
I know it may be too late but my friend suggested that we may use the momentjs library for the date parsing.